### DISCUSSION TITLE
Micropython, RC522 and writing NDEF tags

### DISCUSSION THREAD

**Discussion by maveric** [micropython/micropython#9355] [Libraries & Drivers] [answered]
Are there are libraries out there for reading and writing NDEF tags using micropython from an MCU? I have RC522 setup with an ESP32 (can also use rp2) and can read and write it with no problem, but it doesn't play nice with other readers if I just write text in my own format. But I'm struggling to find a package to help out with NDEF. I've read lots of documentation and understand *what* is happening / needs to happen to read and write tags in the format, but my python skills pretty new and writing a driver like that is still above my head.

I spend several days searching with no luck, then spent the last few days trying to modify:
https://github.com/kichik/pyndef
And even went back a revision before the "six" package was introduced to make sure I wasn't messing something up while making the conversion to not use it. Still no luck.

Is my google fu failing me? Or has this just not been done yet for micropython?

I'm trying to read/write mifare classic 1k cards.

**Comment by maveric** [comment, marked as answer]
I've updated the above library to work with micropython. Tested on rp2 with MP 1.19

https://github.com/maveric/micropython-pyndef

The library handles *only* the NDEF message portion, and not the NDEF container. I handle that at read/write time for now. I plan to update this library to handle the whole process in time.

**Reply by QiaoTuCodes** [reply]
hey guys. rc522 rfid read&write have more demo on here: https://micropython.club/319.html
